independent-projects/tools/platform-descriptor-resolver-json/src/main/java/io/quarkus/platform/descriptor/resolver/json/QuarkusJsonPlatformDescriptorResolver.java

```diff
@@ -711,12 +711,7 @@ private static String getBundledPlatformQuarkusVersionOrNull() {
         } catch (IOException e) {
             throw new IllegalStateException("Failed to load quarkus.properties from the classpath", e);
         }
-        final String quarkusVersion = props.getProperty("plugin-version");
```

Comment:
hmm - isn't there cases where plugin-version is the only thing availalbe as a fallback?

Reply:
It would only fail with an older version of the platform I believe. But we don't have to do backward compat, right @aloubyansky? 

IMHO it's better to be not compatible here, than failing somewhere else because we missed something.

Reply:
you aren't guaranteed to have a platform - even in newer versions.

Reply:
misspoke - or rather, users can make a pom that has no platform ref; but that is not reelvant for here as this is during fully controlled codegenaration.

Reply:
It seems you are mixing with something else right?